### DISCUSSION TITLE
Custom rule loop through any resource type

### DISCUSSION THREAD

**Discussion by tlalmeida** [aws-cloudformation/cfn-lint#2843] [Q&A]
Hi, folks.

Maybe it's not possible but I'd like to know if using custom rules (https://github.com/aws-cloudformation/cfn-lint/blob/main/docs/custom_rules.md) is it possible to just use a `*` to loop through any resource type.

Example: I have a template with multiple resources and all of them must have some required tags.
Is it possible?

As far as I know I have to declare every resource type that I need.

**Comment by kddejong** [comment]
Sorry, I'm way behind on these.  You can use the match function like https://github.com/aws-cloudformation/cfn-lint/blob/dba16cb09e142d89144577aa6505e8871589e2d0/src/cfnlint/rules/resources/properties/Properties.py#L551 and then loop the resources inside the function.

Or you can register for certain types like:
https://github.com/aws-cloudformation/cfn-lint/blob/dba16cb09e142d89144577aa6505e8871589e2d0/src/cfnlint/rules/resources/dynamodb/AttributeMismatch.py#L24
Then this function will be called:
https://github.com/aws-cloudformation/cfn-lint/blob/dba16cb09e142d89144577aa6505e8871589e2d0/src/cfnlint/rules/resources/dynamodb/AttributeMismatch.py#L118 for each of those resource types.
